consensus/core/src/storage/rocksdb_store.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    ops::Bound::{Included, Unbounded},
+    time::Duration,
+};
+
+use consensus_config::AuthorityIndex;
+use typed_store::{
+    metrics::SamplingInterval,
+    reopen,
+    rocks::{default_db_options, open_cf_opts, DBMap, MetricConf, ReadWriteOptions},
+    Map as _,
+};
+
+use super::Store;
+use crate::{
+    block::{BlockDigest, BlockRef, Round, VerifiedBlock},
+    commit::{Commit, CommitIndex},
+    error::ConsensusResult,
+};
+
+/// Persistent storage with RocksDB.
+pub(crate) struct RocksDBStore {
+    /// Stores the blocks by refs.
+    blocks: DBMap<(Round, AuthorityIndex, BlockDigest), bytes::Bytes>,
+    /// A secondary index that orders refs first by authors.
+    digests_by_authorities: DBMap<(AuthorityIndex, Round, BlockDigest), ()>,
+    /// Maps commit index to content.
+    commits: DBMap<CommitIndex, Commit>,
+}
+
+#[allow(unused)]
+impl RocksDBStore {
+    pub(crate) const BLOCKS_CF: &'static str = "blocks";
+    pub(crate) const DIGESTS_BY_AUTHORITIES_CF: &'static str = "digests";
+    pub(crate) const COMMITS_CF: &'static str = "commits";
+
+    /// Creates a new instance of RocksDB storage.
+    pub(crate) fn new(path: &str) -> Self {
```

Comment:
nit: do you mind writing some comments for the intuition for why the initialization values are chosen for rocksdb setup. Specifically for `optimize_db_for_write_throughput` & `optimize_for_large_values_no_scan`